### PR TITLE
test: set ENABLE_MV3 automatically

### LINKED ISSUE #27704: Handle  ENABLE_MV3  flag when run e2e tests on firefox locally
### What is this about?

https://consensys.slack.com/archives/CTQAGKY5V/p1728396101379599

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by HowardBraham**
## **Description**

The stated problem was "automatically set `ENABLE_MV3` based on the browser, in `run-e2e-test`"

However, that would only handle the case of Firefox, and miss builds with:
- `start:mv2`
- `dist:mv2`
- `build:test:flask:mv2`
- `build:test:mv2`
- Any `webpack` build

I had previously written code that reads `manifest.json` from Node, so I thought "hey let's just read it!"  When running a test, you now never even have to **think** about `ENABLE_MV3`.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27748?quickstart=1)

## **Related issues**

Fixes: #27704 

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,8 +25,6 @@
     "build:dev": "node development/build/index.js",
     "start:test": "yarn env:e2e build:dev testDev --apply-lavamoat=false",
     "start:test:flask": "yarn start:test --build-type flask --apply-lavamoat=false",
-    "start:test:mv2:flask": "ENABLE_MV3=false yarn start:test:flask --apply-lavamoat=false --snow=false",
```
Comment (Gudahtt) [inline package.json:28]: Why were these removed?

Comment (HowardBraham) [reply package.json:28]: I could be wrong, but I think these commands are no longer necessary.  I should explicitly test though.

**Comment by HowardBraham** [comment]
@metamaskbot update-policies

File: package.json (hunk 1 above)
Comment (HowardBraham) [reply package.json]: Whoops, you were correct... I put them back in

**Review by chloeYue — DISMISSED** [review]

**Review by pedronfigueiredo — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -135,7 +138,6 @@ Before running e2e tests, ensure you've run `yarn install` to download dependenc
 3. Start a test build with live changes: `yarn start:test` is particularly useful for development. It starts a test build that automatically recompiles application code upon changes. This option is ideal for iterative testing and development. This command also allows you to generate test builds for various types, including:
    - `yarn start:test` for main build
    - `yarn start:test:flask` for flask build
-   - `yarn start:test:mv2` for mv2 build
```
Comment (Gudahtt) [inline README.md:138]: This is still relevant; we're only removing the need for the `ENABLE_MV3` flag for tests here. Not for builds

**Review by Gudahtt — CHANGES_REQUESTED** [review]
Mostly looks good! Just one line removed from the README that I think should stay. I can approve after that is resolved.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by chloeYue — APPROVED** [review]
LGTM.
